MonoGame.Framework/Utilities/Png/PngWriter.cs

```diff
@@ -211,6 +211,16 @@ private void GetColorData(Texture2D texture2D)
 
             switch (texture2D.Format)
             {
+                case SurfaceFormat.Single:
+                    var floatData = new float[colorDataLength];
+                    texture2D.GetData<float>(floatData);
+                    
+                    for (int i = 0; i < colorDataLength; i++) {
+                        // Export as a greyscale image.
+                        colorData[i] = new Color(floatData[i], floatData[i], floatData[i]);
```

Comment:
I don't know if the compiler is smart enough to optimize this down to one array access, but to avoid any extraneous bounds checks, read the array once into a temporary `float` and pass that float into the `Color` constructor.  Using `fixed` would make it faster still, but see if the first approach makes any improvement.

Also, our coding guidelines require the braces to be on the following line at the start of a statement block.